packages/mdc-checkbox/_mixins.scss

```diff
@@ -247,28 +405,42 @@
   }
 }
 
-@mixin mdc-checkbox__background_ {
-  @include mdc-rtl-reflexive-position(
-    left, ($mdc-checkbox-touch-area - $mdc-checkbox-size) / 2, ".mdc-checkbox");
+@mixin mdc-checkbox__background_($query: mdc-feature-all()) {
+  $feat-animation: mdc-feature-create-target($query, animation);
+  $feat-structure: mdc-feature-create-target($query, structure);
+
+  @include mdc-feature-targets($feat-structure) {
+    @include mdc-rtl-reflexive-position(
+      left, ($mdc-checkbox-touch-area - $mdc-checkbox-size) / 2, ".mdc-checkbox");
+
+    display: inline-flex;
+    position: absolute;
+    top: ($mdc-checkbox-touch-area - $mdc-checkbox-size) / 2;
+    bottom: 0;
+    align-items: center;
+    justify-content: center;
+    box-sizing: border-box;
+    width: $mdc-checkbox-ui-pct;
+    height: $mdc-checkbox-ui-pct;
+  }
 
-  display: inline-flex;
-  position: absolute;
-  top: ($mdc-checkbox-touch-area - $mdc-checkbox-size) / 2;
-  bottom: 0;
-  align-items: center;
-  justify-content: center;
-  box-sizing: border-box;
-  width: $mdc-checkbox-ui-pct;
-  height: $mdc-checkbox-ui-pct;
-  transition:
-    mdc-checkbox-transition-exit(background-color),
-    mdc-checkbox-transition-exit(border-color);
-  // border-color is overridden by the mdc-checkbox-unmarked-stroke-color() mixin
-  border: $mdc-checkbox-border-width solid currentColor;
-  border-radius: 2px;
-  background-color: transparent;
-  pointer-events: none;
-  will-change: background-color, border-color;
+  @include mdc-feature-targets($feat-animation) {
+    transition:
+      mdc-checkbox-transition-exit(background-color),
+      mdc-checkbox-transition-exit(border-color);
+  }
+
+  @include mdc-feature-targets($feat-structure) {
+    // border-color is overridden by the mdc-checkbox-unmarked-stroke-color() mixin
+    border: $mdc-checkbox-border-width solid currentColor;
+    border-radius: 2px;
+    background-color: transparent;
+    pointer-events: none;
+  }
+
+  @include mdc-feature-targets($feat-animation) {
```

Comment:
This seems safe to move up and combine with the animation block above. The current separation is presumably due to the order rules in our stylelintrc.